modules/core/core-app/src/main/java/com/enonic/xp/core/impl/app/ApplicationAuditLogSupportImpl.java

```diff
@@ -0,0 +1,115 @@
+package com.enonic.xp.core.impl.app;
+
+import java.net.URL;
+
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+import com.enonic.xp.app.ApplicationKey;
+import com.enonic.xp.audit.AuditLogService;
+import com.enonic.xp.audit.AuditLogUris;
+import com.enonic.xp.audit.LogAuditLogParams;
+import com.enonic.xp.context.Context;
+import com.enonic.xp.context.ContextAccessor;
+import com.enonic.xp.context.ContextBuilder;
+import com.enonic.xp.data.PropertySet;
+import com.enonic.xp.data.PropertyTree;
+import com.enonic.xp.security.PrincipalKey;
+import com.enonic.xp.security.RoleKeys;
+import com.enonic.xp.security.auth.AuthenticationInfo;
+
+@Component(configurationPid = "com.enonic.xp.app")
+public class ApplicationAuditLogSupportImpl
+    implements ApplicationAuditLogSupport
+{
+    private static final String SOURCE = "com.enonic.xp.app";
+
+    private final AuditLogService auditLogService;
+
+    private final AppConfig appConfig;
+
+    @Activate
+    public ApplicationAuditLogSupportImpl( final AppConfig appConfig, @Reference final AuditLogService auditLogService )
+    {
+        this.appConfig = appConfig;
+        this.auditLogService = auditLogService;
+    }
+
+    @Override
+    public void startApplication( final ApplicationKey applicationKey )
+    {
+        if ( appConfig.auditlog_enabled() )
```

Comment:
at could be stored in `volatile boolean` for readability.

Reply:
done